rust-src/id/src/identity_provider.rs

```diff
@@ -271,6 +720,56 @@ pub fn sign_identity_object<
     Ok(ip_secret_key.sign_unknown_message(&message, &mut csprng))
 }
 
+pub fn sign_identity_object<
+    P: Pairing,
+    AttributeType: Attribute<P::ScalarField>,
+    C: Curve<Scalar = P::ScalarField>,
+>(
+    pre_id_obj: &PreIdentityObject<P, C>,
+    ip_info: &IpInfo<P>,
+    alist: &AttributeList<C::Scalar, AttributeType>,
+    ip_secret_key: &ps_sig::SecretKey<P>,
+) -> Result<ps_sig::Signature<P>, Reason> {
+    sign_identity_object_common(&pre_id_obj.common_fields, ip_info, alist, ip_secret_key)
+}
+
+pub fn sign_identity_object_v1<
+    P: Pairing,
+    AttributeType: Attribute<P::ScalarField>,
+    C: Curve<Scalar = P::ScalarField>,
+>(
+    pre_id_obj: &PreIdentityObjectV1<P, C>,
+    ip_info: &IpInfo<P>,
+    alist: &AttributeList<C::Scalar, AttributeType>,
+    ip_secret_key: &ps_sig::SecretKey<P>,
+) -> Result<ps_sig::Signature<P>, Reason> {
+    sign_identity_object_common(&pre_id_obj.common_fields, ip_info, alist, ip_secret_key)
+}
+
+pub fn sign_identity_object_common<
```

Comment:
Since these three functions are very similar, please provide detailed documentation as to what they do (i.e., doc comment), also describing the inputs.